code/game/objects/machinery/telecomms/broadcasting.dm

```diff
@@ -199,6 +199,48 @@
 			continue
 		hearer.Hear(rendered, virt, language, message, frequency, spans)
 
+
+	//Check that speaker isn't banned from TTS
+	var/atom/movable/speaker = virt?.source
+	var/banned = FALSE
+	if(ismob(speaker))
+		var/mob/potential_user = speaker
+		if(is_banned_from(potential_user.ckey, "TTS") || potential_user.client?.prefs.muted & MUTE_TTS)
+			banned = TRUE
+
+	//If they aren't, handle radio TTS
+	if(speaker && speaker.voice && !banned)
+		var/is_speaker_squad_lead = FALSE
+		if(ishuman(speaker))
+			var/mob/living/carbon/human/human_speaker = speaker
+			if(human_speaker.assigned_squad?.squad_leader == speaker)
+				is_speaker_squad_lead = TRUE
+
+		var/is_speaker_command_freq = FALSE
+		if(frequency == FREQ_COMMAND || frequency == FREQ_COMMAND_SOM)
+			is_speaker_command_freq = TRUE
+
+		if(speaker in receive)
+			receive -= speaker //This list isn't used again, so we can just cut out the original speaker from it so TTS doesn't play twice
+
+		var/list/list_of_listeners
+		for(var/mob/living/carbon/human/potential_hearer in receive)
+			if(potential_hearer.stat >= UNCONSCIOUS || potential_hearer.disabilities & DEAF || !(potential_hearer.client?.prefs.sound_tts != TTS_SOUND_OFF))
+				continue
+
+			var/radio_flags = potential_hearer.client.prefs.radio_tts_flags
```

Comment:
```suggestion
			var/radio_flags = potential_hearer?.client?.prefs?.radio_tts_flags
```
Runtime in code/game/objects/machinery/telecomms/broadcasting.dm, line 231: Cannot read null.prefs